src/main/java/org/ossgang/commons/observable/Observables.java

```diff
@@ -78,4 +72,42 @@ private Observables() {
         }
         return combinedProperty;
     }
+
+    /**
+     * Produces an {@link ObservableValue} that on each update of any source {@link ObservableValue} publishes the
+     * result of the combiner applied with the latest values of the other inputs.
+     * The order of the input values of the combiner is the same as the order of the provided source {@link ObservableValue}s
+     *
+     * @param sources  the input {@link ObservableValue}s
+     * @param combiner the combining function that will produce the result
+     * @param <I>      the input type
+     * @param <O>      the output type
+     * @return an {@link ObservableValue} that on each update of any source publishes the result of the
+     * combiner applied with the latest values of the other inputs.
+     */
+    public static <I, O> ObservableValue<O> mergeLatest(List<ObservableValue<I>> sources,
+                                                        Function<List<I>, O> combiner) {
+        Property<O> mergedProperty = property();
+        Map<ObservableValue<I>, I> latestValues = new HashMap<>();
+
+        for (ObservableValue<I> source : sources) {
+            source.subscribe(sourceValue -> {
+                synchronized (latestValues) {
+                    latestValues.put(source, sourceValue);
+
+                    if (latestValues.keySet().containsAll(sources)) {
+                        List<I> latestValueSnapshotInOrder = new ArrayList<>();
+                        for (ObservableValue<I> s : sources) {
+                            latestValueSnapshotInOrder.add(latestValues.get(s));
+                        }
+
+                        mergedProperty.set(combiner.apply(latestValueSnapshotInOrder));
+                    }
+                }
+            }, FIRST_UPDATE);
+        }
+
+        return mergedProperty;
```

Comment:
it is not a property, only an observableValue?

Reply:
it will be always a property since we need the set to propagate the values :)

Reply:
Mhmmm ... really? I thought we use ObservableValue on purpose, as we only propagate downstream ... (Of course we might inherit from DerivedObservableValue and use the `dispatchValue` method ... however, there should be no `set` method, I think ...

In my view, `Property` indicates somthing bidirectional: It can be set from downstream! ... (e.g. by using mechanics like `bindBidi` or `mapBidi` (which still have to be implemented. ... However, I think a bidi-connection with a zip operator is for sure not possible (or at least too complicated that it would be worth implementing ;-)

Reply:
yes, indeed the return value is an ObservableValue :) We don't want to expose the Property that we use internally.

Also, yes I agree there is no bidirectional binding available for zip and combine :)